### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by edurdevic**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,12 +2,29 @@
...
-Eg.
 
-* [Maintenance operations](#maintenance-operations) (vacuum, optimize, cleanup, etc.)
-* [Scan & classify](#scan--classify) content for patterns (phone numbers, emails, etc.)
-* Data transformations based on semantic types
+## Operations examples
+
+Operations are applied concurrently across multiple tables
+
+* Manitenance operations
```
Comment (david-tempelmann) [inline README.md:13]: Manitenance -> typo

File: README.md [hunk 2]
```diff
@@ -24,209 +41,46 @@ from discoverx import DX
...
-```
-dx.from_tables("default.*.*")\
-  .apply_sql("VACUUM {full_table_name}")\
-  .explain()
-```
+You can now run operations across multiple tables. 
 
-You can also filter tables that have a specific column name. 
-Eg.
+For example you can select one row from all tables which name contains `sample` in a catalog starting with `dev_` with
```
Comment (david-tempelmann) [inline README.md:46]: I found the sentence not to easy to follow. ChatGpt thought that the following formulation would be more clear :-)

> As an illustration, consider the scenario where you need to retrieve a single row from various tables within a catalog that begins with "dev_" and includes the term "sample" in their names. To achieve this, the following code block utilizes the dx.from_tables function, which applies an SQL query to extract JSON-formatted data:

File: README.md [hunk 3]
```diff
@@ -2,12 +2,29 @@
...
 
+![Multi-table operations](docs/images/DiscoverX_Multi-table_operations.png)
+
 Run a single command to execute operations across many tables. 
-Eg.
 
-* [Maintenance operations](#maintenance-operations) (vacuum, optimize, cleanup, etc.)
-* [Scan & classify](#scan--classify) content for patterns (phone numbers, emails, etc.)
-* Data transformations based on semantic types
+## Operations examples
```
Comment (david-tempelmann) [inline README.md:9]: Is the following list ordered somehow w.r.t. importance or relevance. I was just thinking about semantic classification, which in the beginning was our main target and we list it almost in the end. Any thought behind this?

File: README.md [hunk 4]
```diff
@@ -24,209 +41,46 @@ from discoverx import DX
...
+The functions available from the `DataExplorer` Object are:
 
-### Custom rules
+* `having_columns` restricts the selection to tables that have the specified columns
+* `with_concurrency` defines how many queries are executed concurrently (10 by defailt)
+* `apply_sql` applies a SQL template to all tables and returns a [DataExplorerActions](#dataexploreractions-object) object
+* `unpivot_string_columns` returns a melted (unpivoted) dataframe with all string columns from the selected tables and returns a [DataExplorerActions](#dataexploreractions-object) object
 
-You can provide your custom scanning rules based on regex expressions.
+### DataExplorerActions Object
```
Comment (david-tempelmann) [inline README.md:75]: Same as with DataExplorer Object ...

File: docs/Arbitrary_multi-table_SQL.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+
+## Arbitrary SQL template execution
+
+You can run arbitrary SQL operations on multiple tables. In order for this to work, the dataset of each SQL command should return the same schema. You can align the schemas by:
+* Running schema-agnostic commands (Eg. Vacuum, Optimize, History, Describe, etc.)
+* Specifying the common column names to be selected
+* Melting multiple columns into a "long" format
```
Comment (david-tempelmann) [inline docs/Arbitrary_multi-table_SQL.md:7]: Could we provide an example of this?

File: README.md (hunk 3 above)
Comment (edurdevic) [reply README.md:9]: I ordered them by how often I think they can be used.
Semantic classification is very powerful, but maintenance operations is something that everyone can/should do.

File: README.md [hunk 6]
```diff
@@ -24,209 +41,46 @@ from discoverx import DX
...
+* `from_tables("<catalog>.<schema>.<table>")` selects tables based on the specified pattern (use `*` as a wildcard). Returns a [DataExplorer](#dataexplorer-object) object. 
+* `intro` gives an introduction to the library
+* `scan` scans the lakehouse with regex expressions defined by the rules and to power the semantic classification. [Documentation](docs/Semantic_classification.md)
+* `display_rules` shows the rules available for semantic classification
+* `search` searches the lakehouse content for by leveraging the semantic classes identified with scan (eg. email, ip address, etc.). [Documentaiton](docs/Search.md)
+* `select_by_class` selects data from the lakehouse content by semantic class. [Documentation](docs/Select_by_class.md)
+* `delete_by_class` deletes from the lakehouse by semantic class. [Documentation](docs/Delete_by_class.md)
 
-You can define 
+### DataExplorer Object
```
[the remark this answers is not in the document]
Comment (edurdevic) [reply README.md:66]: I moved this into the previous list as a sub-list. Does this make more sense?

File: docs/Arbitrary_multi-table_SQL.md (hunk 5 above)
Comment (edurdevic) [reply docs/Arbitrary_multi-table_SQL.md:7]: Hmm, maybe we replace this with the to_json example. We have a dedicated function for the string melt.

File: README.md (hunk 4 above)
Comment (edurdevic) [reply README.md:75]: Nesting this section would get it too complex, check out how it looks now after I moved the DataExplorer docs.

Comment (david-tempelmann) [reply README.md:71]: I think it's fine to have it here but I would describe it more on a high level and not refer to `DataExplorerActions`. Rather describe HOW you can use `explain` and `execute`, i.e. that you can chain these and give a very short example. It's not important at this point which object they belong to as this makes it a bit too complex I think.

File: docs/Arbitrary_multi-table_SQL.md (hunk 5 above)
Comment (david-tempelmann) [reply docs/Arbitrary_multi-table_SQL.md]: yeah that would make sense

**Review by david-tempelmann — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (edurdevic) [reply README.md:71]: Ok, got it! 
Then let's merge this as it is, and I will open a new one with this change.

**Pull request merged**